src/core/public/application/types.ts

```diff
@@ -222,6 +284,12 @@ export interface InternalApplicationSetup {
    */
   register(plugin: PluginOpaqueId, app: App): void;
 
+  /**
+   * TODO
+   * @param statusUpdater$
+   */
+  registerAppStatusUpdater(statusUpdater$: Observable<AppStatusUpdater>): void;
```

Comment:
Could make `InternalApplicationSetup extends Pick<ApplicationSetup, 'registerAppStatusUpdater'>` to reduce the boilerplate here.